### PR TITLE
Feature/getting time intervals to step by step

### PR DESCRIPTION
**Pull request by JanosevicRa177**

### CODE REVIEW

File: src/HospitalLibrary/Appointments/AppointmentService.cs [hunk 1]
```diff
@@ -27,6 +28,26 @@ public Task<IEnumerable<Appointment>> GetAll()
             return _unitOfWork.AppointmentRepository.GetAll();
         }
 
+        public async Task<IEnumerable<TimeInterval>> GetTimeIntervalsForStepByStep(int doctorId, DateTime chosen)
+        {
+            WorkingHours doctorsWorkingHours = _unitOfWork.WorkingHoursRepository.GetOne((int)chosen.DayOfWeek,doctorId);
```
Comment (srdjan1210) [inline src/HospitalLibrary/Appointments/AppointmentService.cs:33]: I think DayOfWeek is original data type, so why passing the value as int?

File: src/HospitalLibrary/Appointments/AppointmentService.cs [hunk 2]
```diff
@@ -27,6 +28,26 @@ public Task<IEnumerable<Appointment>> GetAll()
...
+            List<TimeInterval> possibleTimeIntervals = new List<TimeInterval>();
+            TimeSpan timeIntervalFiller = doctorsWorkingHours.Start;
+            TimeSpan timeSpanIncrementer = TimeSpan.FromMinutes(30);
+            while (TimeSpan.Compare(timeIntervalFiller, doctorsWorkingHours.End) < 0)
+            {
+                TimeInterval possibleTimeInterval = new TimeInterval(chosen, timeIntervalFiller,
+                    timeIntervalFiller.Add(timeSpanIncrementer));
+                timeIntervalFiller = timeIntervalFiller.Add(timeSpanIncrementer);
+                if (await _intervalValidation.IsIntervalOverlapingWithDoctorAppointments(doctorId,
```
Comment (srdjan1210) [inline src/HospitalLibrary/Appointments/AppointmentService.cs:42]: Remove await from if directly and store it in variable, this is not readable

File: src/HospitalLibrary/Shared/Validators/TimeIntervalValidationService.cs [hunk 3]
```diff
@@ -18,7 +18,19 @@ public TimeIntervalValidationService(IUnitOfWork unitOfWork)
         {
             _unitOfWork = unitOfWork;
         }
-        
+
+        public async Task<bool> IsIntervalOverlapingWithDoctorAppointments(int doctorId,TimeInterval possibleTimeInterval)
+        {
+            IEnumerable<TimeInterval> doctorsAppointmentsTimeIntervals = (await _unitOfWork.AppointmentRepository
+                .GetAllDoctorTakenIntervalsForDate(doctorId,possibleTimeInterval.Start));
+            foreach (TimeInterval timeInterval in doctorsAppointmentsTimeIntervals)
```
Comment (srdjan1210) [inline src/HospitalLibrary/Shared/Validators/TimeIntervalValidationService.cs:26]: I think there is a LINQ expression that can write this much shorter.
Something like: doctorAppointmentTimeIntervals.Any(app => possibleTimeInterval.IsOverlaping(app));

**Review by srdjan1210 — CHANGES_REQUESTED** [review]

File: src/HospitalLibrary/Appointments/AppointmentService.cs (hunk 1 above)
Comment (JanosevicRa177) [reply src/HospitalLibrary/Appointments/AppointmentService.cs:34]: DayOfWeek  is type itself not int, so cant be changed

**Review by srdjan1210 — APPROVED** [review]

**Pull request merged**